components/style/restyle_hints.rs

```diff
@@ -383,38 +405,79 @@ impl DependencySet {
     }
 
     pub fn clear(&mut self) {
-        self.deps.clear();
+        self.common_deps.clear();
+        self.attr_deps.clear();
+        self.state_deps.clear();
     }
 }
 
 impl DependencySet {
+    fn test_dependency<E>(&self,
```

Comment:
This doesn't really make sense as a Dep on DependencySet. Can we impl it as a method on Vec<Dependency>, or does rust not let us do that? If not, I guess best would be to make it a non-method helper function.